### ISSUE TITLE
problem to learn model at 2 GPU's | device=[0,1]

### ISSUE THREAD

**Issue by Antonoof** [ultralytics/ultralytics#22015]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hello, in the new version of ultralytics, I have encountered the problem of training on two GPUs. There is no such problem in the old version. Can you tell me if there are new hyperparameters or a new way to parallelize the process?

<img width="1095" height="781" alt="Image" src="https://github.com/user-attachments/assets/f09a21bb-27ae-4b9b-aced-e265d261ea1f" />

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Antonoof, thank you for your interest in Ultralytics 🚀! This is an automated response to help get things moving quickly. An Ultralytics engineer will also assist you here soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report (e.g., regression from an older version), please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it. Helpful details include:
- Exact command(s) used (e.g., `yolo train ... device=0,1`)
- `ultralytics` version and install method
- OS, CUDA, PyTorch versions, and GPU models
- Full training logs/tracebacks and a short snippet of your config or overrides

If this is a custom training ❓ Question (e.g., how to run multi-GPU in the latest release), please provide as much information as possible, including the exact CLI/Python command you’re using, screenshots/logs, and confirm you’re following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/). This will help us clarify any changes in distributed/multi-GPU usage across versions.

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade
Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:
```bash
pip install -U ultralytics
```

## Environments
YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>
If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by glenn-jocher** [comment]
Thanks for the report—multi‑GPU usage hasn’t changed in YOLO11; use `device=0,1` (CLI) or `device=[0,1]` (Python), set an explicit integer `batch` divisible by GPU count (AutoBatch `batch<1` is not supported on DDP), and avoid `rect=True`. See the Multi‑GPU examples in the Train docs and GPU troubleshooting tips here: [Train mode: Multi‑GPU](https://docs.ultralytics.com/modes/train/) • [Troubleshooting GPU setup](https://docs.ultralytics.com/guides/yolo-common-issues/).

Python:
```python
from ultralytics import YOLO
model = YOLO("yolo11n.pt")
model.train(data="coco8.yaml", epochs=100, imgsz=640, device=[0, 1], batch=32)
```

CLI:
```bash
yolo detect train model=yolo11n.pt data=coco8.yaml device=0,1 batch=32
```

If it still fails, please paste the exact command, full text traceback/logs (not screenshots), and your system info from `yolo checks` (or `!yolo checks` in notebooks); if you recently upgraded in Jupyter, restart the kernel.

**Comment by UltralyticsAssistant** [comment]
A potential fix has been merged in PR [Unwrap DDP model before loss calculation](https://github.com/ultralytics/ultralytics/pull/22016) by @Y-T-G with contributions from @Laughing-q. This change ensures the training loss is computed on the underlying model by unwrapping any wrappers first, replacing `self.model.loss(...)` with `unwrap_model(self.model).loss(...)`. This improves compatibility with torch.compile and wrappers like DDP/EMA that may hide or alter the `.loss` method, reducing runtime errors without changing expected training behavior or APIs.

To test whether this resolves the issue:
- Install the latest code from main:
  `pip install -U "git+https://github.com/ultralytics/ultralytics.git@main"`
- Alternatively, wait for the next official PyPI release and then update:
  `pip install -U ultralytics`

Feedback on whether the merged changes resolve the problem would be appreciated. Thanks for reporting this and please share any remaining issues so they can be addressed promptly. 🙏

**Comment by Laughing-q** [comment]
@Antonoof thanks for spotting! We recently have some issue with our server so lacking ddp tests for now(we'll try to have it back asap), and didn't notice recent changes broke the ddp workflow. The issue should be resolved in this PR: https://github.com/ultralytics/ultralytics/pull/22016. Sorry for the inconvenience. :)
